### PR TITLE
feat(rome_js_semantic): simple globals resolution

### PR DESCRIPTION
**Pull request by xunilrj**
## Summary

This PR implements a simple global resolution for the semantic model.

## Test Plan

```
> cargo test -p rome_js_semantic -- ok_semantic_model_globals
```

### CODE REVIEW

File: crates/rome_js_semantic/src/semantic_model.rs [hunk 1]
```diff
@@ -996,19 +1050,33 @@ impl SemanticModelBuilder {
...
+            global_references: self.global_references,
         };
         SemanticModel::new(data)
     }
 }
 
+#[derive(Default)]
+/// Extra options for the [SemanticModel] creation.
+pub struct SemanticModelOptions {
+    /// All the allowed globals names
```
Comment (ematipico) [inline crates/rome_js_semantic/src/semantic_model.rs:1064]:
```suggestion
    /// All the allowed global names
```

**Review by ematipico — APPROVED** [review]

File: crates/rome_js_semantic/src/semantic_model.rs [hunk 2]
```diff
@@ -996,19 +1050,33 @@ impl SemanticModelBuilder {
...
+    pub globals: HashSet<String>,
```
Comment (leops) [inline crates/rome_js_semantic/src/semantic_model.rs:1065]: How does `HashSet` compare in performance against using a pre-sorted `Vec` of strings and a binary search ?

File: crates/rome_js_semantic/src/semantic_model.rs [hunk 3]
```diff
@@ -1041,7 +1109,7 @@ mod test {
             FileId::zero(),
             SourceType::js_module(),
         );
-        let model = semantic_model(&r.tree());
+        let model = semantic_model(&r.tree(), SemanticModelOptions::default());
```
[the remark this answers is not in the document]
Comment (xunilrj) [reply crates/rome_js_semantic/src/semantic_model.rs:1112]: > Shouldn't we fill the globals with the ones that we have inside globals/mod.rs? 

Yes, but it should be done at crates\rome_js_analyze\src\lib.rs:76 where you have info from the file being analyzed and can choose the globals.

File: crates/rome_js_semantic/src/semantic_model.rs (hunk 2 above)
Comment (xunilrj) [reply crates/rome_js_semantic/src/semantic_model.rs:1065]: I prefer to merge this like it is, and we can bench in a real case later. I did a similar test while ago tough: https://github.com/rome/tools/pull/2975

**Pull request merged**